src/compute/src/render/join/mz_join_core.rs

```diff
@@ -457,14 +468,14 @@ where
                         cursor1.step_val(storage1);
                         cursor2.rewind_vals(storage2);
 
-                        *fuel = fuel.saturating_sub(temp.len());
+                        *work += temp.len();
```

Comment:
We should use a `saturating_add` here to protect against overflows (I think we could trigger this with zero-sized types.)

Reply:
Good point, fixed!